### PR TITLE
DOC: Update release instructions

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/development/maintaining.rst [hunk 1]
```diff
@@ -449,9 +449,13 @@ which will be triggered when the tag is pushed.
     git tag -a v1.5.0.dev0 -m "DEV: Start 1.5.0"
     git push upstream main --follow-tags
 
-3. Build the source distribution (git must be in the tag commit)::
+3. Download the source distribution and wheels from the `wheel staging area <https://anaconda.org/scientific-python-nightly-wheels/pandas>`_.
+   Be careful to make sure that no wheels are missing (e.g. due to failed builds).
```
Comment (mroeschke) [inline doc/source/development/maintaining.rst:453]: Would be good to describe how we can tell if a build failed

File: doc/source/development/maintaining.rst [hunk 2]
```diff
@@ -463,23 +467,19 @@ which will be triggered when the tag is pushed.
    - Set as the latest release: Leave checked, unless releasing a patch release for an older version
      (e.g. releasing 1.4.5 after 1.5 has been released)
 
-5. The GitHub release will after some hours trigger an
+5. Upload wheels to PyPI::
+
+    twine upload pandas/dist/pandas-<version>*.{whl,tar.gz} --skip-existing
```
Comment (mroeschke) [inline doc/source/development/maintaining.rst:472]: Still not fond of using the pypi publish action https://github.com/pypa/gh-action-pypi-publish (by gating it behind a manual trigger)?

File: doc/source/development/maintaining.rst (hunk 1 above)
Comment (lithomas1) [reply doc/source/development/maintaining.rst:453]: You kind of have to look manually for the individually jobs unfortunately. 

I'm trying to build reporting for failed jobs here
https://github.com/pandas-dev/pandas/issues/55903

File: doc/source/development/maintaining.rst (hunk 2 above)
Comment (mroeschke) [reply doc/source/development/maintaining.rst:472]: Nice yeah I think that would work. I can also try to look into this sometime in the future

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @lithomas1

File: scripts/download_wheels.sh [hunk 3]
```diff
@@ -11,6 +11,7 @@
 # one by one to the dist/ directory where they would be generated.
 
 VERSION=$1
+mkdir -p $(dirname -- $0)/../dist
```
Comment (datapythonista) [inline scripts/download_wheels.sh:14]: Sorry for the late review. Small comment, but seems like you could just do `mkdir -p $DIST_DIR` if you write this in the line after defining the directory, no?

File: doc/source/development/maintaining.rst (hunk 2 above)
Comment (jorisvandenbossche) [reply doc/source/development/maintaining.rst:472]: The latest version of the GitHub action now also implements the "Trusted Publisher" model of PyPI (https://github.com/marketplace/actions/pypi-publish#trusted-publishing), avoiding creating an API token that is stored as a secret in github

File: scripts/download_wheels.sh (hunk 3 above)
Comment (lithomas1) [reply scripts/download_wheels.sh:14]: You're right, I think.

Will make a note to do this (hopefully I don't forget).

**Pull request merged**